editor/src/clj/editor/material.clj

```diff
@@ -566,27 +587,29 @@ there is no way a user can know what the generated id will be for older shaders.
   (output save-value g/Any produce-save-value)
   (output build-targets g/Any :cached produce-build-targets)
   (output shader ShaderLifecycle :cached produce-shader)
-  (output samplers [g/KeywordMap] (gu/passthrough samplers))
+  (output samplers [g/KeywordMap] :cached produce-samplers)
   (output attribute-infos [g/KeywordMap] :cached produce-attribute-infos))
 
 (defn- legacy-texture->sampler [name]
   (assoc default-pb-sampler :name name))
 
-(defn load-material [_project self resource material-desc]
+(defn load-material [project self resource material-desc]
   {:pre [(map? material-desc)]} ; Material$MaterialDesc in map format.
   (let [resolve-resource #(workspace/resolve-resource resource %)
         attributes->editable-attributes #(mapv attribute->editable-attribute %)]
-    (gu/set-properties-from-pb-map self Material$MaterialDesc material-desc
-      vertex-program (resolve-resource :vertex-program)
-      fragment-program (resolve-resource :fragment-program)
-      vertex-constants (render-program-utils/constants->editable-constants :vertex-constants)
-      fragment-constants (render-program-utils/constants->editable-constants :fragment-constants)
-      attributes (attributes->editable-attributes :attributes)
-      name :name
-      samplers (render-program-utils/samplers->editable-samplers :samplers)
-      tags :tags
-      vertex-space :vertex-space
-      max-page-count :max-page-count)))
+    (concat
+      (g/connect project :default-sampler-filter-modes self :default-sampler-filter-modes)
+      (gu/set-properties-from-pb-map self Material$MaterialDesc material-desc
+        vertex-program (resolve-resource :vertex-program)
+        fragment-program (resolve-resource :fragment-program)
+        vertex-constants (render-program-utils/constants->editable-constants :vertex-constants)
+        fragment-constants (render-program-utils/constants->editable-constants :fragment-constants)
+        attributes (attributes->editable-attributes :attributes)
+        name :name
+        samplers (render-program-utils/samplers->editable-samplers :samplers)
+        tags :tags
+        vertex-space :vertex-space
+        max-page-count :max-page-count))))
```

Comment:
Indentation change from adding the `:default-sampler-filter-modes` connection from the `Project` node.